src/MySqlConnector/MySqlClient/MySqlConnection.cs

```diff
@@ -280,19 +295,10 @@ private void DoClose()
 		{
 			if (m_connectionState != ConnectionState.Closed)
 			{
-				m_cachedProcedures = null;
-				if (ActiveReader != null){
-					ActiveReader.Dispose();
-					ActiveReader = null;
-				}
-				if (CurrentTransaction != null)
-				{
-					CurrentTransaction.Dispose();
-					CurrentTransaction = null;
-				}
+				CloseDatabase();
 				if (m_session != null)
 				{
-					if (m_connectionSettings.Pooling)
+					if (m_connectionSettings.Pooling && m_changedDatabase == null)
```

Comment:
Once I actually *ran* the connection pooling tests, they immediately failed (because the pool was completely exhausted).